src/Microsoft.Azure.WebJobs.Extensions.Dapr/Services/DaprServiceListener.cs

```diff
@@ -187,12 +187,22 @@ internal async Task WarnIfSidecarMisconfigured()
 
                 try
                 {
-                    int port = appConnectionProperties.GetProperty("port").GetInt32();
-                    if (port != appPort)
+                    if (appConnectionProperties.TryGetProperty("port", out var port))
                     {
-                        this.logger.LogWarning($"The Dapr sidecar is configured to listen on port {port}, but the app server is running on port {appPort}. This may cause unexpected behavior, see https://aka.ms/azfunc-dapr-app-config-error.");
+                        // port is an int in the Metadata API response.
+                        var portInt = port.GetInt32();
+                        if (portInt != appPort)
+                        {
+                            this.logger.LogWarning($"The Dapr sidecar is configured to listen on port {portInt}, but the app server is running on port {appPort}. This may cause unexpected behavior, see https://aka.ms/azfunc-dapr-app-config-error.");
+                        }
+                    }
+                    else
+                    {
+                        // Daprd sidecar does not have port configured.
```

Comment:
this looks good! The link does not seem to work for me

Reply:
Yes, it's a TODO, that doc needs to be created.